Add fitText field to block.json schema

aditya241104 opened the pull request:
## What
Adds the `fitText` block support definition to the JSON schema.

## Why  
The `fitText` block support was recently added in #71904, but the schema definition was missing. This adds proper validation and autocomplete support for developers using the fitText feature.

## How
- Added `fitText` field to the `supports.properties` object in the block.json schema
- Followed alphabetical ordering consistent with other block supports
- Added appropriate description and type definitions

## Testing Instructions
✅ Verified JSON schema syntax is valid
✅ Confirmed fitText property is now recognized in block.json files
✅ All existing tests continue to pass

Fixes #72190

aditya241104 committed:
Add fitText field to block.json schema

Adds fitText block support definition to enable proper validation
and autocomplete for the fitText feature added in #71904.

Fixes #72190

t-hamano reviewed (commented):
Thanks for the PR!

The feature should be placed at the `supports.typography,fitText`, not the `supports.fitText`. In other words, this feature should be placed here:

https://github.com/WordPress/gutenberg/blob/955c993d25ba5771d2612be8adeef3b27cdfe550/schemas/json/block.json#L616

aditya241104 committed:
Move fitText to supports.typography.fitText location

Addresses review feedback to place fitText under typography
supports rather than as a top-level support property.

t-hamano reviewed (approved):
LGTM! Thanks for your first contribution!

<img width="1345" height="330" alt="Image" src="https://github.com/user-attachments/assets/9db7ae12-22ca-4456-8708-784e2b12c78e" />
